lib/win32/certstore/store_base.rb

```diff
@@ -102,6 +102,29 @@ def cert_delete(store_handler, certificate_thumbprint)
         cert_delete_flag
       end
 
+      # Search certificate from open certificate store and return boolean
+      # store_handler => Open certificate store handler
+      # search_token => CN, RDN or any certificate attribute
+      def cert_search(store_handler, search_token)
+        validate!(search_token)
+        cert_rdn = memory_ptr
+        certificate_list = Hash.new
+        counter = 0
+        begin
+          while (pcert_context = CertEnumCertificatesInStore(store_handler, pcert_context) and !pcert_context.null?)
+            cert_property = get_cert_property(pcert_context)
+            if cert_property.include?(search_token)
+              key = "certificate#{counter = counter + 1}"
```

Comment:
This feels odd. Why are we creating a complex key name here? Could we use an array instead of a hash and `Array.push` each certificate into it?